webrender/src/render_backend.rs

```diff
@@ -566,3 +556,91 @@ impl GLContextDispatcher for WebRenderGLDispatcher {
         dispatcher.as_mut().unwrap().as_mut().unwrap().dispatch(f);
     }
 }
+
+pub enum BackendMsg {
+    NewRenderBackend(RenderBackendInit),
+    DeleteRenderBackend(RendererId),
+}
+
+pub struct RenderBackendThread {
+    api_tx: MsgSender<(ApiMsg, RendererId)>,
+    init_tx: Sender<BackendMsg>,
+    next_renderer_id: RendererId,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum BackendStatus {
+    Continue,
+    Remove,
+}
+
+impl RenderBackendThread {
+    pub fn new(thread_name: String) -> Result<Self, ::std::io::Error> {
+        let (api_tx, api_rx) = try! { msg_channel() };
+        let (init_tx, init_rx) = channel();
+
+        try!{ thread::Builder::new().name(thread_name.clone()).spawn(move || {
+
+            register_thread_with_profiler(thread_name);
+
+            let mut render_backends = HashMap::new();
+
+            while let Ok((api_msg, id)) = api_rx.recv() {
+
+                while let Ok(init_msg) = init_rx.try_recv() {
```

Comment:
This looks a bit wonky to me. How about having a separate channel for each backend instance, so that this code waits for all {backend1, backend2, ..., backendN, init} channels at once?